### PR TITLE
fix a bug where editing causing an error lost the original form

### PR DESCRIPTION
**Pull request by czue**
@dannyroberts - encountered this looking into something else

### CODE REVIEW

File: corehq/ex-submodules/couchforms/util.py [hunk 1]
```diff
@@ -496,6 +496,10 @@ def process(xform):
                             # log the error and respond with a success code so that the phone doesn't
                             # keep trying to send the form
                             instance = _handle_known_error(e, instance)
+                            xforms[0] = instance
+                            # this is usually just one document, but if an edit errored we want
+                            # to save the deprecated form as well
+                            XFormInstance.get_db().bulk_save(xforms)
```
Comment (czue) [inline corehq/ex-submodules/couchforms/util.py]: before just the `instance` was saved, and the newly added deprecated form in the context just vanished.

Comment (dannyroberts) [inline corehq/ex-submodules/couchforms/util.py]: Oh, dear

**Comment by dannyroberts** [comment]
The failure looks possibly real?
```
======================================================================
FAIL: testBadIndexReferenceDomain (casexml.apps.case.tests.test_indexes.IndexTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "corehq/ex-submodules/casexml/apps/case/tests/test_indexes.py", line 163, in testBadIndexReferenceDomain
    self.assertIsInstance(xform, XFormError)
AssertionError: XFormInstance(_attachments={'form.xml': {'data': u'PGRhdGEgeG1sbnM9Imh0dHBzOi8vd3d3LmNvbW1jYXJlaHEub3JnL3Rlc3QvY2FzZXhtbC13cmFwcGVyIiB4bWxuczpqcm09Imh0dHA6Ly9vcGVucm9zYS5vcmcvanIveGZvcm1zIj48Y2FzZSBjYXNlX2lkPSJjaGlsZC1jYXNlLWlkIiBkYXRlX21vZGlmaWVkPSIyMDE1LTA3LTA2IiB1c2VyX2lkPSJ0ZXN0LWluZGV4LXVzZXIiIHhtbG5zPSJodHRwOi8vY29tbWNhcmVocS5vcmcvY2FzZS90cmFuc2FjdGlvbi92MiI+PGNyZWF0ZT48Y2FzZV90eXBlIC8+PGNhc2VfbmFtZSAvPjxvd25lcl9pZCAvPjwvY3JlYXRlPjxpbmRleD48YmFkIGNhc2VfdHlwZT0iYmFkLWNhc2UiPnRleHQtaW5kZXgtbW90aGVyLWNhc2U8L2JhZD48L2luZGV4PjwvY2FzZT48L2RhdGE+', 'content_type': u'text/xml'}}, _id=u'e70d1201d08336e4580d508ce13c1c9d', _rev=u'1-8b58d7e91924da3a08f2d007c8ec7757', app_id=None, auth_context={'doc_type': u'DefaultAuthContext'}, build_id=None, computed_={}, computed_modified_on_=None, date_header=None, doc_type=u'XFormError', domain=u'child', export_tag=[u'domain', u'xmlns'], form={u'@xmlns': u'https://www.commcarehq.org/test/casexml-wrapper', u'case': {u'@xmlns': u'http://commcarehq.org/case/transaction/v2', u'index': {u'bad': {u'@case_type': u'bad-case', u'#text': u'text-index-mother-case'}}, u'@date_modified': datetime.date(2015, 7, 6), u'create': {u'case_type': u'', u'owner_id': u'', u'case_name': u''}, u'@user_id': u'test-index-user', u'@case_id': u'child-case-id'}, '#type': u'data'}, history=[], initial_processing_complete=False, last_sync_token=None, openrosa_headers={}, partial_submission=False, path=None, received_on=datetime.datetime(2015, 7, 6, 18, 45, 18, 60136), submit_ip=None, xmlns=u'https://www.commcarehq.org/test/casexml-wrapper', problem=u'IllegalCaseId: Bad case id') is not an instance of <class 'couchforms.models.XFormError'>
----------------------------------------------------------------------
```

**Comment by czue** [comment]
yep, will fix

**Comment by czue** [comment]
:white_check_mark: @dannyroberts

**Comment by czue** [comment]
@TylerSheffels (since @dannyroberts is offline today)

**Pull request merged**